src/victory-util/data.js

```diff
@@ -63,34 +64,45 @@ export default {
    * @returns {Array} the formatted data
    */
   formatData(dataset, props, stringMap) {
-    if (!Array.isArray(dataset)) {
+    const isArrayOrIterable = Array.isArray(dataset) || Immutable.isIterable(dataset);
+    if (!isArrayOrIterable) {
       return [];
     }
+
     stringMap = stringMap || {
       x: this.createStringMap(props, "x"),
       y: this.createStringMap(props, "y")
     };
+
     const accessor = {
       x: Helpers.createAccessor(props.x !== undefined ? props.x : "x"),
       y: Helpers.createAccessor(props.y !== undefined ? props.y : "y"),
       y0: Helpers.createAccessor(props.y0 !== undefined ? props.y0 : "y0")
     };
-    const data = dataset.map((datum, index) => {
+
+    const data = dataset.reduce((dataArr, datum, index) => { // eslint-disable-line complexity
```

Comment:
using `reduce` here because js arrays and `immutable` `List`s both have `reduce` and we can then return a regular array.